### PR TITLE
Scripts to automate autosploit.

### PR DESCRIPTION
**Pull request by Selora**
Wrappers around autosploit.
Could be use to test against a whitelist of your organisation's IP addresses. 

**./dryrun_autosploit.sh whitelist.txt "search_query"**
Will search censys/shodan/etc and do a dry-run against discovered hosts that are in the whitelist.
This will generate a report containing the dry-run results for every hosts in the whitelist.
This does NOT exploit anything.

**./run_autosploit.sh**
Will run autosploit in exploit mode against previously discovered hosts in the whitelist.
This will generate a report with MSF results for every hosts in the whitelist.

IMPORTANT: 
- You **should** do a dry-run before running "run_autosploit.sh".
- Make sure that the whitelist is sound with a given scope, and validate against the dry-run report.
- In other words, **VALIDATE THE DRYRUN REPORT BEFORE LAUNCHING THE EXPLOIT RUN.**

The code to compare each hosts against a whitelist **hasn't been thoroughly tested**. If you don't validate a dryrun report and run the "run_autosploit.sh" script, and you hit hosts that are outside your given scope, you have been warned.

Also added a Vagrant file. Could be useful to someone, I don't intend to push this a something functional or usable out-of-the-box. Feel free to drop it.

### CODE REVIEW

File: etc/json/other_modules.json [hunk 1]
```diff
@@ -0,0 +1,268 @@
+{
```
Comment (Ekultek) [inline etc/json/other_modules.json]: why?

File: Vagrant/Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Use as a strating point to spin up a box in lightsail.
```
Comment (Ekultek) [inline Vagrant/Vagrantfile]: Need a Vagrant run in the README, set that up real quick

File: etc/json/other_modules.json (hunk 1 above)
Comment (Selora) [reply etc/json/other_modules.json]: This is a list of non-blocking modules. Without using this module file, autosploit is not usable against multiple host since most fuzzers are hanging for an indefinite amount of time.

Comment (Ekultek) [reply etc/json/other_modules.json]: So it's basically the same as `default_modules.json` without fuzzers correct?

File: Vagrant/Vagrantfile (hunk 2 above)
Comment (Selora) [reply Vagrant/Vagrantfile]: Can I leave this to you? I'm just putting this as-is, I do not intend to make it useful nor to explain my setup. 
That what I used as a dev environment, I'm just putting this in case somebody would be interested.

Comment (Ekultek) [reply Vagrant/Vagrantfile]: I have no idea how vagrant works, so we'll just add an issue about it I guess

File: etc/json/other_modules.json (hunk 1 above)
Comment (Selora) [reply etc/json/other_modules.json]: Yes

Comment (Ekultek) [reply etc/json/other_modules.json]: So why wouldn't we just fix the fuzzers instead of creating a duplicate file?

Comment (Selora) [reply etc/json/other_modules.json]: It's a dev branch; I'm just putting a file I'm able to work with, without deleting stuff some other might rely on. 
Feel free to modify the default_modules.json and drop the other_modules.json file.

Comment (Ekultek) [reply etc/json/other_modules.json]: yeah just delete this and I'll merge the vagrant

**Comment by NullArray** [comment]
Perhaps we can just put the fuzzers in a seperate JSON file and ship it with AutoSploit. Then simply add an option to run AutoSploit in "fuzzer mode", instead of users having to provide their own JSON file, this way we don't have to get rid of the fuzzers and we'll be user friendly by providing an easy solution.

**Comment by Selora** [comment]
I'm going to close this PR and rebase these commits into something cleaner.
